azurelinuxagent/common/osutil/gaia.py

```diff
@@ -202,7 +202,7 @@ def set_hostname(self, hostname):
     def set_dhcp_hostname(self, hostname):
         logger.warn('set_dhcp_hostname is ignored on GAiA')
 
-    def publish_hostname(self, hostname):
+    def publish_hostname(self, hostname, recover_nic=False):
```

Comment:
why do we set to False explicitly?

Reply:
We don't want to have to change the call to publish_hostname in provisioning. If we don't set recover_nic to False explicitly, then we'll need to add another argument to the call in provisioning

Reply:
never mind, this is an override so you need to make it consistent with the definition in the base class